hclsyntax/parser_test.go

```diff
@@ -2197,6 +2194,7 @@ block "valid" {}
 					"a": {
 						Name: "a",
 						Expr: &LiteralValueExpr{
+							Val: cty.DynamicVal,
```

Comment:
The previous approach to this test was, it turns out, not actually checking the `Val` field at all, because `cty.Value` has only unexported fields.

Switching to `cmp` brought attention to that first by requiring the use of `ctydebug.CmpOptions` (which teaches `cmp` how to compare two `cty.Value` values) and then by showing failures for this test and the other similar one below.